core/domain/summary_services.py

```diff
@@ -63,6 +63,16 @@ def get_displayable_exp_summary_dicts_matching_ids(exploration_ids):
     displayable_exp_summaries = []
     exploration_summaries = (
         exp_services.get_exploration_summaries_matching_ids(exploration_ids))
+    displayable_exp_summaries = get_displayable_exp_summary_dicts(
+        exploration_summaries, exploration_ids)
+
+    return displayable_exp_summaries
+
+def get_displayable_exp_summary_dicts(exploration_summaries, exploration_ids):
```

Comment:
I don't think this should bring in the exploration_ids, since it duplicates the data in exploration_summaries, and we should try to pass as little data as possible between functions. Instead, have the first line of this function be

    exploration_ids = [exploration_summary.id for exploration_summary in exploration_summaries]